src/callback.rs

```diff
@@ -244,6 +244,23 @@ where
     let unwind_safe_py = AssertUnwindSafe(pool.python());
     let panic_result = panic::catch_unwind(move || -> PyResult<_> {
         let py = *unwind_safe_py;
+        #[cfg(all(PyPy, not(Py_3_8)))]
+        {
+            const PYPY_GOOD_VERSION: [u8; 3] = [7, 3, 8];
+            let version = py
+                .import("sys")?
+                .getattr("implementation")?
+                .getattr("version")?;
+            if version.compare(crate::types::PyTuple::new(py, &PYPY_GOOD_VERSION))?
+                == std::cmp::Ordering::Less
+            {
+                let warn = py.import("warnings")?.getattr("warn")?;
+                warn.call1((
+                    "PyPy 3.7 versions older than 7.3.8 are known to have binary \
+                             compatibility issues which may cause segfaults. Please upgrade.",
+                ))?;
+            }
+        }
```

Comment:
I think this is the one place where we're changing the behavior of the code on pypy3.7. I don't have an obvious reason this would cause a crash, but it does jump out to me that we're doing this import+operations on every single function invocation.

Reply:
...huh?

It looks like this piece was committed to main with https://github.com/PyO3/pyo3/pull/2217 but then picked into this branch with https://github.com/PyO3/pyo3/pull/2262/commits/c18e60f92894173e7ff6fa1ce0d6ecd002ebc824 but put in a different location...

Reply:
> I don't have an obvious reason this would cause a crash,

I think it's most likely that this set an exception/warning when it shouldn't have. For example, if you raise[^1] a warning during `__traverse__`, you can segfault.


[^1]: `__traverse__` doesn't give you a Python token, so this requires unsafe, which makes this not a problem